presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -372,7 +372,8 @@ PlanNodeStatsEstimate addJoinComplementStats(
             SymbolStatsEstimate joinComplementSymbolStats = joinComplementStats.getSymbolStatistics(symbol);
 
             // weighted average
-            double newNullsFraction = (innerJoinSymbolStats.getNullsFraction() * innerJoinRowCount + joinComplementSymbolStats.getNullsFraction() * joinComplementRowCount) / outputRowCount;
+            double newNullsFraction = outputRowCount == 0 ? 1 :
```

Comment:
> I think you should bailout just after

I can bail out if the complement rows count is zero by just returning the inner join stats.

> But in all honesty I think CBO should simply return unknown stats when output row count drops to 0 because proper queries shoudn't really produce no results

That seems to be a bigger change that has to be contemplated carefully.